### ISSUE TITLE
Kotest tests with @MicronautTest and empty constructor are skipped

### ISSUE THREAD

**Issue by adamkobor** [micronaut-projects/micronaut-test#902]
### Expected Behavior

We had a few tests where we had `@MicronautTest` over the test class, but the tests were actually not relying on Micronaut, so we didn't even have a single constructor argument in the test class that should have been injected. This isn't completely cool of course, because we didn't need the annotation on the test class, but all those tests worked just fine before upgrading to micronaut-test 4.1.0. 
I would expect that these tests would not be skipped after the upgrade

### Actual Behaviour

The tests were skipped completely.

### Steps To Reproduce

Create a Kotest test class, annotate it with @MicronautTest, but don't inject anything from Micronaut (e.g. leave the constructor empty)

### Environment Information

Stack: Kotest5, micronaut-test 4.1.0, Kotlin 1.9.20
MacOS arm64

### Version

4.2.0

**Comment by MT-Jacobs** [comment]
This is a legitimate issue. It hits particularly hard for any project that uses Kotest `AnnotationSpec`s that rely on `@Injected` params.

Only workaround is to inject an unused bean in each test's constructor.

**Comment by sdelamo** [comment]
do you have a sample app?

**Comment by adamkobor** [comment]
> do you have a sample app?

no, because it's really easy to reproduce

**Comment by sdelamo** [comment]
> > do you have a sample app?
> 
> no, because it's really easy to reproduce

If it is straightforward to reproduce, it would be easy to create a sample app. Creating a sample app helps address issues faster.

**Comment by MT-Jacobs** [comment]
Here's a sample app. https://github.com/MT-Jacobs/issue902

@adamkobor I did it this way if you ever want to try creating samples yourself:
```
sdk upgrade micronaut # Upgrades to latest version of micronaut
mn create-app -l=kotlin -t=kotest -b=gradle_kotlin issue902
```
From there I just copy-pasted the base test to make a failing test.

**Comment by MT-Jacobs** [comment]
Here's that failing test -

https://github.com/MT-Jacobs/issue902/blob/main/src/test/kotlin/issue902/EmpyConstructorTest.kt#L5-L11

**Comment by timyates** [comment]
Weird...

[We have a test for this](https://github.com/micronaut-projects/micronaut-test/blob/74e1efcd32fc73b40a1ce10625bea5a28e1cbca9/test-kotest5/src/test/kotlin/io/micronaut/test/kotest5/EmptyConstructorTest.kt), and as far as I can currently tell, that test is executed 🤔

**Comment by timyates** [comment]
Ok, [found why it's being ignored](https://github.com/micronaut-projects/micronaut-test/blob/2b3c5ef893de9c9f76f87848065cbe38af5594ea/test-kotest5/src/main/kotlin/io/micronaut/test/extensions/kotest5/MicronautKotest5Extension.kt#L38-L39), now to work on a fix

**Comment by lmachacek** [comment]
We ran into a problem where the constructor based workaround does not help and it is in a situation, when you need to use `TestPropertyProvider`, which does not work with constructor based DI. 
See the example: https://github.com/lmachacek/micronaut-test-property-provider
Is there any other way how to overcome this situation?

**Comment by timyates** [comment]
@lmachacek That looks like a different issue 🤔 

Could you raise a separate issue including details and the reproducer project

PRs welcome btw, as I'm in no way a Kotest expert

**Comment by lmachacek** [comment]
In the meantime I found another workaround for `FunSpec` based tests, which can be also used in combination with `TestPropertyProvider`. And it is only about changing of `test` to `context`, for example `test("This is ignored") { ... }` to `context("This is not ignored") { ... }`.
And I also found out that this _ignore problem_ wasn't in Micronaut 4.1.6 (Kotest 4.6.2) but has been introduced in 4.2.x (Kotest 4.7.2) probably by some change in Kotest. 
I'll try find more time to look on it :wink:

**Comment by lmachacek** [comment]
From my observations it seems that this PR https://github.com/kotest/kotest/pull/3649 caused that this function is not called https://github.com/micronaut-projects/micronaut-test/blob/2b3c5ef893de9c9f76f87848065cbe38af5594ea/test-kotest5/src/main/kotlin/io/micronaut/test/extensions/kotest5/MicronautKotest5Extension.kt#L62 anymore. 
Thanks to that `specDefinition` remains empty and then all tests are ignored.

Fortunately this fixed that https://github.com/kotest/kotest/pull/3716

So, to sum it up, it worked with Kotest up to 5.6.2 and then all 5.7.x versions weren't working well with Micronaut tests and fortunately Kotest 5.8.0 is OK again :wink: 

Solution for me is to force Kotest dependency to version 5.8.0 instead of 5.7.2 provided by Micronaut.

**Comment by lmachacek** [comment]
Now I can see that Micronaut 4.3.0 with Kotest 5.8.0 dependency was released recently. So, probably this issue can be closed as _Fixed in Micronaut 4.3.0_ :wink:

**Comment by timyates** [comment]
@lmachacek Thank you so much for the help! 😎 

This is indeed fixed by the Kotest bump to 5.8.0 that comes with Micronaut 4.3.0

@adamkobor The fix here is to upgrade either Micronaut, or Kotest (Micronaut would be best)
